### PR TITLE
Determine fee payer once the transaction review is shown

### PR DESCRIPTION
**Pull request by GhenadieVP**
Determine the fee payer when after the transaction review screen is shown, this is meant to reduce the time it takes the Wallet to analyze and present the Transaction Review.

# What was done:
- Extract the selection of default fee payer in a separate function on TransactionClient.
- Which in turn is called from TransactionReview after the transaction was analysed.
- While the default fee payer is determined we do show a shimmer animation in TransactionFee section.

# Demo

https://github.com/radixdlt/babylon-wallet-ios/assets/118184705/6f18ceda-d00d-45bf-a208-b2a5dda4be91

### CODE REVIEW

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview+View.swift [hunk 1]
```diff
@@ -30,9 +30,10 @@ extension TransactionReview.State {
 			viewControlState: viewControlState,
 			rawTransaction: displayMode.rawTransaction,
 			showApprovalSlider: reviewedTransaction != nil,
-			canApproveTX: canApproveTX && reviewedTransaction?.feePayingValidation == .valid,
+			canApproveTX: canApproveTX && reviewedTransaction?.feePayingValidation.wrappedValue == .valid,
 			sliderResetDate: sliderResetDate,
-			canToggleViewMode: reviewedTransaction != nil && reviewedTransaction?.transaction != .nonConforming
+			canToggleViewMode: reviewedTransaction != nil && reviewedTransaction?.transaction != .nonConforming,
+			viewRawTransactionButtonState: reviewedTransaction?.feePayer.isSuccess == true ? .enabled : .disabled
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview+View.swift:36]: Disable toggling to view raw transaction until the fee payer status is determined.

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift [hunk 2]
```diff
@@ -486,6 +484,33 @@ public struct TransactionReview: Sendable, FeatureReducer {
...
+				reviewedTransaction.transactionFee = selectionResult.updatedFee
+				reviewedTransaction.transactionSigners = selectionResult.transactionSigners
+				reviewedTransaction.signingFactors = selectionResult.signingFactors
+			}
+
+			state.reviewedTransaction = reviewedTransaction
+			state.networkFee?.reviewedTransaction = reviewedTransaction
+
+			if reviewedTransaction.transaction == .nonConforming {
+				return showRawTransaction(&state)
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift:506]: Refresh the raw transaction view after the fee payer was determined.

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift [hunk 3]
```diff
@@ -486,6 +484,33 @@ public struct TransactionReview: Sendable, FeatureReducer {
...
+			}
+			return .none
+
+		case let .determineFeePayerResult(.failure(error)):
+			assertionFailure("Failed to determine fee payer \(error)")
+			state.reviewedTransaction?.feePayer = .success(nil)
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift:512]: Failure is handled as no fee payer could be determined.

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift [hunk 4]
```diff
@@ -611,6 +636,27 @@ extension TransactionReview {
 			throw FailedToAddGuarantee(underlyingError: error)
 		}
 	}
+
+	func determineFeePayer(_ state: State, reviewedTransaction: ReviewedTransaction) -> Effect<Action> {
+		if reviewedTransaction.transactionFee.totalFee.lockFee == .zero {
+			.send(.internal(.determineFeePayerResult(.success(nil))))
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift:642]: No lockFee => no fee payer needed.

File: RadixWallet/Clients/TransactionClient/Models/TransactionFee.swift
Comment (GhenadieVP) [inline RadixWallet/Clients/TransactionClient/Models/TransactionFee.swift:1]: Just moved to a separate file

File: RadixWalletTests/Clients/TransactionClientTests/TransactionClientTests.swift [hunk 5]
```diff
@@ -150,13 +150,15 @@ final class TransactionClientTests: TestCase {
 			}
 		} operation: {
 			try await TransactionClient.feePayerSelectionAmongstCandidates(
+				request: .init(
+					networkId: .enkinet,
```
Comment (CyonAlexRDX) [inline RadixWalletTests/Clients/TransactionClientTests/TransactionClientTests.swift:154]: Maybe update test to use mainnet? Looks better...

**Review by CyonAlexRDX — APPROVED** [review]
LGTM !

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift [hunk 6]
```diff
@@ -1180,52 +1229,54 @@ enum FeeValidationOutcome {
...
-			return .valid
+				return .valid
+			}
 		}
 	}
 }
 
-extension FeePayerSelectionAmongstCandidates {
-	var validate: FeeValidationOutcome {
+extension FeePayerCandidate? {
```
Comment (Sajjon) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift:1267]: perhaps straining for type-checker to spell `FeePayerCandidate?`, maybe `extension Optional<FeePayerCandidate>` is faster? more clear as well, I think I prefer it.

**Pull request merged**